<a href="https://colab.research.google.com/github/Ihenetu-Victoria09/Victoria_Tag_Along_Stage_A/blob/main/Victoria's_Tag_along_proj.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
csv_df = pd.read_csv("/Users/Victoria Ihenetu/Desktop/FoodBalanceSheets_E_Africa_NOFLAG(4).csv")
csv_df.to_csv("/Users/Victoria Ihenetu/Desktop/FoodBalanceSheets_E_Africa_NOFLAG(4).csv")
csv_df.head()

Unnamed: 0  Unnamed: 0.1  Area Code     Area  Item Code         Item  \
0           0             0          4  Algeria       2501   Population   
1           1             1          4  Algeria       2501   Population   
2           2             2          4  Algeria       2901  Grand Total   
3           3             3          4  Algeria       2901  Grand Total   
4           4             4          4  Algeria       2901  Grand Total   

   Element Code                                 Element             Unit  \
0           511           Total Population - Both sexes     1000 persons   
1          5301                Domestic supply quantity      1000 tonnes   
2           664           Food supply (kcal/capita/day)  kcal/capita/day   
3           674  Protein supply quantity (g/capita/day)     g/capita/day   
4           684      Fat supply quantity (g/capita/day)     g/capita/day   

      Y2014     Y2015     Y2016     Y2017     Y2018  
0  38924.00  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00      0.00  
2   3377.00   3379.00   3372.00   3341.00   3322.00  
3     94.90     94.35     94.72     92.82     91.83  
4     80.06     79.36     77.40     80.19     77.28

In [ ]:
FBS_df = pd.read_csv("/Users/Victoria Ihenetu/Desktop/FoodBalanceSheets_E_Africa_NOFLAG(4).csv")
FBS_df.dropna

<bound method DataFrame.dropna of        Unnamed: 0  Unnamed: 0.1  Unnamed: 0.1.1  Area Code      Area  \
0               0             0               0          4   Algeria   
1               1             1               1          4   Algeria   
2               2             2               2          4   Algeria   
3               3             3               3          4   Algeria   
4               4             4               4          4   Algeria   
...           ...           ...             ...        ...       ...   
60938       60938         60938           60938        181  Zimbabwe   
60939       60939         60939           60939        181  Zimbabwe   
60940       60940         60940           60940        181  Zimbabwe   
60941       60941         60941           60941        181  Zimbabwe   
60942       60942         60942           60942        181  Zimbabwe   

       Item Code           Item  Element Code  \
0           2501     Population           511   
1  

In [ ]:
FBS_df = pd.read_csv("/Users/Victoria Ihenetu/Desktop/FoodBalanceSheets_E_Africa_NOFLAG(4).csv")
FBS_df.to_csv("/Users/Victoria Ihenetu/Desktop/FoodBalanceSheets_E_Africa_NOFLAG(4).csv", index = False)
FBS_df.head(10)

Unnamed: 0  Unnamed: 0.1  Unnamed: 0.1.1  Area Code     Area  Item Code  \
0           0             0               0          4  Algeria       2501   
1           1             1               1          4  Algeria       2501   
2           2             2               2          4  Algeria       2901   
3           3             3               3          4  Algeria       2901   
4           4             4               4          4  Algeria       2901   
5           5             5               5          4  Algeria       2903   
6           6             6               6          4  Algeria       2903   
7           7             7               7          4  Algeria       2903   
8           8             8               8          4  Algeria       2941   
9           9             9               9          4  Algeria       2941   

               Item  Element Code                                 Element  \
0        Population           511           Total Population - Both sexes   
1        Population          5301                Domestic supply quantity   
2       Grand Total           664           Food supply (kcal/capita/day)   
3       Grand Total           674  Protein supply quantity (g/capita/day)   
4       Grand Total           684      Fat supply quantity (g/capita/day)   
5  Vegetal Products           664           Food supply (kcal/capita/day)   
6  Vegetal Products           674  Protein supply quantity (g/capita/day)   
7  Vegetal Products           684      Fat supply quantity (g/capita/day)   
8   Animal Products           664           Food supply (kcal/capita/day)   
9   Animal Products           674  Protein supply quantity (g/capita/day)   

              Unit     Y2014     Y2015     Y2016     Y2017     Y2018  
0     1000 persons  38924.00  39728.00  40551.00  41389.00  42228.00  
1      1000 tonnes      0.00      0.00      0.00      0.00      0.00  
2  kcal/capita/day   3377.00   3379.00   3372.00   3341.00   3322.00  
3     g/capita/day     94.90     94.35     94.72     92.82     91.83  
4     g/capita/day     80.06     79.36     77.40     80.19     77.28  
5  kcal/capita/day   2932.00   2958.00   2941.00   2921.00   2932.00  
6     g/capita/day     67.14     67.38     67.37     66.11     67.10  
7     g/capita/day     53.84     54.72     52.04     55.59     54.57  
8  kcal/capita/day    444.00    421.00    431.00    421.00    390.00  
9     g/capita/day     27.76     26.97     27.35     26.71     24.73

# Total aggregate sum of Animal fat produced in the year 2014 and 2017:
# 2014 = 209460.54
# 2017 = 269617.53

In [ ]:
FBS_df.groupby('Item')['Y2017'].sum()

Item
Alcohol, Non-Food        2348.00
Alcoholic Beverages     95581.06
Animal Products         11547.65
Animal fats            269617.53
Apples and products     10198.90
                         ...    
Vegetables, Other      157752.59
Vegetal Products       107655.20
Wheat and products     240047.62
Wine                     4178.02
Yams                   229174.59
Name: Y2017, Length: 119, dtype: float64

# From the whole dataset the mean and the standard deviation for the year 2015 is: 
# mean in 3 decimal places =135.24
# Standrad deviation in 3 decimal places = 1603.404


In [ ]:
FBS_df = pd.read_csv("/Users/Victoria Ihenetu/Desktop/FoodBalanceSheets_E_Africa_NOFLAG(4).csv")
FBS_df.describe(include= "all")


Unnamed: 0  Unnamed: 0.1  Unnamed: 0.1.1     Area Code   Area  \
count   60943.000000  60943.000000    60943.000000  60943.000000  60943   
unique           NaN           NaN             NaN           NaN     49   
top              NaN           NaN             NaN           NaN  Kenya   
freq             NaN           NaN             NaN           NaN   1560   
mean    30471.000000  30471.000000    30471.000000    134.265576    NaN   
std     17592.873065  17592.873065    17592.873065     72.605709    NaN   
min         0.000000      0.000000        0.000000      4.000000    NaN   
25%     15235.500000  15235.500000    15235.500000     74.000000    NaN   
50%     30471.000000  30471.000000    30471.000000    136.000000    NaN   
75%     45706.500000  45706.500000    45706.500000    195.000000    NaN   
max     60942.000000  60942.000000    60942.000000    276.000000    NaN   

           Item Code                     Item  Element Code  \
count   60943.000000                    60943  60943.000000   
unique           NaN                      119           NaN   
top              NaN  Milk - Excluding Butter           NaN   
freq             NaN                     1262           NaN   
mean     2687.176706                      NaN   3814.856456   
std       146.055739                      NaN   2212.007033   
min      2501.000000                      NaN    511.000000   
25%      2562.000000                      NaN    684.000000   
50%      2630.000000                      NaN   5142.000000   
75%      2775.000000                      NaN   5511.000000   
max      2961.000000                      NaN   5911.000000   

                         Element         Unit          Y2014          Y2015  \
count                      60943        60943   59354.000000   59395.000000   
unique                        18            5            NaN            NaN   
top     Domestic supply quantity  1000 tonnes            NaN            NaN   
freq                        5295        40933            NaN            NaN   
mean                         NaN          NaN     134.196282     135.235966   
std                          NaN          NaN    1567.663696    1603.403984   
min                          NaN          NaN   -1796.000000   -3161.000000   
25%                          NaN          NaN       0.000000       0.000000   
50%                          NaN          NaN       0.090000       0.080000   
75%                          NaN          NaN       8.340000       8.460000   
max                          NaN          NaN  176405.000000  181137.000000   

                Y2016          Y2017          Y2018  
count    59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN  
top               NaN            NaN            NaN  
freq              NaN            NaN            NaN  
mean       136.555222     140.917765     143.758381  
std       1640.007194    1671.862359    1710.782658  
min      -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000  
50%          0.080000       0.100000       0.070000  
75%          8.430000       9.000000       9.000000  
max     185960.000000  190873.000000  195875.000000

In [ ]:
FBS_df.isnull().sum()

Unnamed: 0           0
Unnamed: 0.1         0
Unnamed: 0.1.1       0
Area Code            0
Area                 0
Item Code            0
Item                 0
Element Code         0
Element              0
Unit                 0
Y2014             1589
Y2015             1548
Y2016             1535
Y2017             1506
Y2018             1436
dtype: int64

In [ ]:

import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib

In [ ]:
#plt.figure(figsize=(7,4))
#plt.xticks(rotation=90)
#plot = pd.DataFrame(FBS_df)
#sns.barplot(data=plot, x = 'unit', y = 'count')
#plt.xlabel(plot)

In [ ]:
#sns.barplot(FBS_df, x='unit', y='count')
#plt.show()

# The year 2017 has the highest sum of Import Quantity

In [ ]:
FBS_df.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

# The total sum of production for the year 2014 = 1931287.75

In [ ]:
FBS_df.groupby('Element')['Y2014'].sum()

# In year 2018, Domestic supply quantity has the highest sum = 2161192.10

In [ ]:
FBS_df.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

# In year 2018, Protein supply quantity (g/capita/day) is the third to the lowest 

In [ ]:
FBS_df.groupby('Element')['Y2018'].sum()

Element
Domestic supply quantity                  2161192.10
Export Quantity                            181594.80
Fat supply quantity (g/capita/day)          10258.69
Feed                                       233489.68
Food                                      1303841.28
Food supply (kcal/capita/day)              455261.00
Food supply quantity (kg/capita/yr)         49056.85
Import Quantity                            287997.09
Losses                                     163902.00
Other uses (non-food)                       91300.97
Processing                                 308429.00
Production                                2075072.89
Protein supply quantity (g/capita/day)      11833.56
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Total Population - Both sexes             1140605.00
Tourist consumption                            90.00
Name: Y2018, dtype: float64

In [ ]:
print(FBS_df['Area'])

0         Algeria
1         Algeria
2         Algeria
3         Algeria
4         Algeria
           ...   
60938    Zimbabwe
60939    Zimbabwe
60940    Zimbabwe
60941    Zimbabwe
60942    Zimbabwe
Name: Area, Length: 60943, dtype: object


In [ ]:
FBS_df.groupby('Element')['Area'].count()

Element
Domestic supply quantity                  5295
Export Quantity                           4403
Fat supply quantity (g/capita/day)        5023
Feed                                      1319
Food                                      4941
Food supply (kcal/capita/day)             5014
Food supply quantity (kg/capita/yr)       4905
Import Quantity                           5139
Losses                                    2009
Other uses (non-food)                     1732
Processing                                2010
Production                                3881
Protein supply quantity (g/capita/day)    5023
Residuals                                 4655
Seed                                       762
Stock Variation                           4232
Total Population - Both sexes               45
Tourist consumption                        555
Name: Area, dtype: int64

In [ ]:
FBS_df.groupby('Area'),('Element)['Import Qunatity'].first()

SyntaxError: invalid syntax (Temp/ipykernel_15944/3579956178.py, line 1)

60943


# Total Number of unique countries is 49

In [ ]:
FBS_df.groupby('Area')['Area'].count()

Area
Algeria                        1313
Angola                         1424
Benin                          1285
Botswana                       1378
Burkina Faso                   1409
C?te d'Ivoire                  1467
Cabo Verde                     1260
Cameroon                       1460
Central African Republic       1344
Chad                           1343
Comoros                          45
Congo                          1413
Djibouti                       1095
Egypt                          1488
Eswatini                       1284
Ethiopia                       1447
Ethiopia PDR                     39
Gabon                          1245
Gambia                         1168
Ghana                          1354
Guinea                         1410
Guinea-Bissau                  1128
Kenya                          1560
Lesotho                        1150
Liberia                        1206
Madagascar                     1439
Malawi                         1447
Mali                   